### ISSUE TITLE
shmget: No space left on device

### ISSUE THREAD

**Issue by atlesn** [atlesn/rrr#19]
```
Nov 11 00:57:09 label-print-01 rrr[13636]: Error from shmget in __rrr_mmap_channel_allocate: No space left on device
Nov 11 00:57:09 label-print-01 rrr[13636]: Could not allocate memory in rrr_mmap_channel_write
Nov 11 00:57:09 label-print-01 rrr[13636]: Could not send address message on mmap channel in __rrr_cmodule_send_message name
Nov 11 00:57:09 label-print-01 rrr[13636]: Error while sending message in rrr_cmodule_send_to_fork
```

```
------ Shared Memory Limits --------
max number of segments = 4096
max seg size (kbytes) = 18014398509465599
max total shared memory (kbytes) = 18014398509481980
min seg size (bytes) = 1
```

All modules restarted correctly afterwards.

**Comment by atlesn** [comment]
Seems like when the MMAP channel in some cases allocates a lot of shmget segments. These should either get cleaned up using some algorithm, or there must be a custom allocator to reduce the number of segments.

**Comment by atlesn** [comment]
The previous commit introduced a bug causing buffer overflow, fixed with 5b80ac2528bf07e7664934c6617c8a87199336df .

**Comment by atlesn** [comment]
Closing this for now, the constant sending of PING/PONG messages cleans up shm-allocated blocks over time thus preventing hitting the default OS limit. Another solution might still be needed if a lot of cmodules are run simultaneously.
